src/d/semantic/caster.d

```diff
@@ -650,3 +673,86 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 		return CastKind.Invalid;
 	}
 }
+
+version(unittest) {
+	BuiltinType[] floatTypes =
+		[BuiltinType.Float, BuiltinType.Double, BuiltinType.Real];
+	BuiltinType[] signedTypes =
+		[BuiltinType.Byte, BuiltinType.Short, BuiltinType.Int, BuiltinType.Long,
+		 BuiltinType.Cent];
+	BuiltinType[] unsignedTypes =
+		[BuiltinType.Ubyte, BuiltinType.Ushort, BuiltinType.Uint,
+		 BuiltinType.Ulong, BuiltinType.Ucent];
+	void assertCastKind(const CastKind ck, const CastKind shouldBe) {
```

Comment:
Considering the same pattern repeats every time, I'd have this helper function instead:

```d
void checkBuiltinCast(BuiltinType from, BuiltinType to, CastKind implicit, CastKind explicit) {
   // ...
}
```